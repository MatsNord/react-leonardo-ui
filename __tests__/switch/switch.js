import React from 'react';
import renderer from 'react-test-renderer';

import Switch from '../../src/switch';

test('Normal switch renders correctly', () => {
  const tree = renderer.create(
    <Switch checked />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch with states renders correctly', () => {
  const tree = renderer.create(
    <Switch checked active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Switch checked variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch passes props correctly', () => {
  const onChange = jest.fn(() => {});

  const tree = renderer.create(
    <Switch checked className="my-class" data-x="x" title="Title" onChange={onChange} />
  ).toJSON();

  const input = tree.children[0].children[0];
  input.props.onChange();
  expect(onChange.mock.calls.length).toBe(1);
});
