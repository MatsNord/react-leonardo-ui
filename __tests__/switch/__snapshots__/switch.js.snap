exports[`test Normal switch renders correctly 1`] = `
<div
  className="lui-switch"
  title={undefined}>
  <label
    className="lui-switch__label"
    htmlFor={undefined}>
    <input
      checked={true}
      className="lui-switch__checkbox"
      type="checkbox" />
    <span
      className="lui-switch__wrap">
      <div
        className="lui-switch__inner" />
      <div
        className="lui-switch__switch" />
    </span>
  </label>
</div>
`;

exports[`test Switch with modifiers renders correctly 1`] = `
<div
  className="lui-switch lui-switch--inverse"
  title={undefined}>
  <label
    className="lui-switch__label"
    htmlFor={undefined}>
    <input
      checked={true}
      className="lui-switch__checkbox"
      type="checkbox" />
    <span
      className="lui-switch__wrap">
      <div
        className="lui-switch__inner" />
      <div
        className="lui-switch__switch" />
    </span>
  </label>
</div>
`;

exports[`test Switch with states renders correctly 1`] = `
<div
  className="lui-switch lui-active lui-disabled"
  title={undefined}>
  <label
    className="lui-switch__label"
    htmlFor={undefined}>
    <input
      checked={true}
      className="lui-switch__checkbox"
      type="checkbox" />
    <span
      className="lui-switch__wrap">
      <div
        className="lui-switch__inner" />
      <div
        className="lui-switch__switch" />
    </span>
  </label>
</div>
`;
